testCloseUnassignedFailedJobAndStopUnassignedStoppingDatafeed test failure

Build link: https://gradle-enterprise.elastic.co/s/sbvr6vobv5ely

```
org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT > testCloseUnassignedFailedJobAndStopUnassignedStoppingDatafeed FAILED
    java.lang.AssertionError: expected:<opened> but was:<opening>
        at __randomizedtesting.SeedInfo.seed([A40942F407E8D4B5:B0C129096422217]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.lambda$setupJobAndDatafeed$14(MlDistributedFailureIT.java:439)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:892)
        at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.setupJobAndDatafeed(MlDistributedFailureIT.java:436)
        at org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.testCloseUnassignedFailedJobAndStopUnassignedStoppingDatafeed(MlDistributedFailureIT.java:214)
```

Reproduction line: 
```
./gradlew ':x-pack:plugin:ml:internalClusterTest' --tests "org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.testCloseUnassignedFailedJobAndStopUnassignedStoppingDatafeed" -Dtests.seed=A40942F407E8D4B5 -Dtests.security.manager=true -Dtests.locale=et -Dtests.timezone=Libya -Dcompiler.java=13
```

Looking at the logs, it seems as if the `.ml-state-*` index takes 15+ seconds to move from yellow to green state. This shouldn't cause problems for the job starting as we only look at primary shards. 

It has only failed once, and it might be due to system resource constraints. Will mute if more failures occur.

--- a/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java
+++ b/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/integration/MlDistributedFailureIT.java
@@ -437,7 +437,7 @@ public class MlDistributedFailureIT extends BaseMlIntegTestCase {
             GetJobsStatsAction.Response statsResponse =
                     client().execute(GetJobsStatsAction.INSTANCE, new GetJobsStatsAction.Request(job.getId())).actionGet();
             assertEquals(JobState.OPENED, statsResponse.getResponse().results().get(0).getState());
-        }, 20, TimeUnit.SECONDS);
+        }, 30, TimeUnit.SECONDS);
 
         setMlIndicesDelayedNodeLeftTimeoutToZero();
 

